### PR TITLE
Feature/enhancements

### PR DESCRIPTION
**Pull request by EdwinFajardoBarrera**
- docker-compose.yml
- application.properties -> application.yml
- Database connection
- JTW first draft
- Register endpoint
- Error handling

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,39 +1,39 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-starter-parent</artifactId>
-        <version>2.4.5</version>
+        <version>2.7.14</version>
```
Comment (EdwinFajardoBarrera) [inline pom.xml:9]: Updated to a supported version

File: pom.xml [hunk 2]
```diff
@@ -1,39 +1,39 @@
...
-  <groupId>com.encora-mid.app</groupId>
-  <artifactId>encora-mid-office</artifactId>
-  <packaging>jar</packaging>
-  <version>1.0-SNAPSHOT</version>
-  <name>encora-mid-office</name>
-  <url>http://maven.apache.org</url>
-  <properties>
+    </parent>
+
+    <groupId>com.encora.office.app</groupId>
```
Comment (EdwinFajardoBarrera) [inline pom.xml:13]: Refactor to follow naming conventions

File: pom.xml [hunk 3]
```diff
@@ -115,10 +113,22 @@
                             <artifactId>mapstruct-processor</artifactId>
                             <version>${org.mapstruct.version}</version>
                         </path>
+                        <path>
+                            <groupId>org.projectlombok</groupId>
```
Comment (EdwinFajardoBarrera) [inline pom.xml:117]: Added loombok

File: src/main/java/com/encora/office/app/config/JwtRequestFilter.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    private UserService userService;
+    // private AuthService authService;
+
+    @Autowired
+    public JwtRequestFilter(UserService userService, AuthService authService) {
+        this.userService = userService;
+        // this.authService = authService;
+    }
+
+    // @Autowired
```
Comment (EdwinFajardoBarrera) [inline src/main/java/com/encora/office/app/config/JwtRequestFilter.java:44]: TODO - Remove these comments

File: src/main/java/com/encora/office/app/config/SecurityConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    // Exception {
+    // http.csrf().disable();
+    // return http.build();
+    // }
+
+    @Bean
+    protected DefaultSecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        log.debug("Filter chain");
+
+        http.sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS).and()
```
Comment (EdwinFajardoBarrera) [inline src/main/java/com/encora/office/app/config/SecurityConfiguration.java:33]: Authorization filter based on JWT filter

**Review by onemargaro — APPROVED** [review]
LGTM, great improvemente and jwt implementation, lets merge it and continue with this asap

**Pull request merged**